main.go

```diff
@@ -184,6 +186,17 @@ var head = []string{
 	"<!doctype", // HTML doctype
 }
 
+func getExtension(s string) string {
+	// Get the base filename for parsing
+	filename := filepath.Base(s)
+	splitStr := strings.Split(filename, ".")
+	if len(splitStr) == 1 {
+		return splitStr[0]
+	}
+	// Reappend the `.` to the extension
+	return strings.Join([]string{".", splitStr[len(splitStr)-1]}, "")
```

Comment:
How about just using existing methods without splitting strings manually. For instance:

```go
func fileExtension(name string) string {
  if v := filepath.Ext(s); v != "" {
    return v
  }
  return filepath.Base(name)
}
```

Much more clear, easier to read imho.

You might also want to pass it to `strings.ToLower()` before returning the result.

Reply:
I like this approach. I was also of the opinion that my approach was not very clean to read so I'll use this instead.